### PR TITLE
feature: Privacy Center - configure email xor phone to match fulfillment functionality

### LINKED ISSUE #2538: Privacy Center allows for both email and phone number to be used in current state.
### Bug Description

Privacy Center allows for email and phone to be used in current state. Currently, it is an XOR. 

#### Steps to Reproduce

1. Open Privacy Center
2. Add a phone
3. Add an email
4. Submit request

### Expected behavior

It disables one field of the two fields to be xor 

### Screenshots

_If applicable, add screenshots to help explain your problem._

### Environment

* Version: 2.6.3
* OS: OSX

### PR DESCRIPTION
**Pull request by nicolas-ethyca**
Closes #2538 

### Code Changes

* [ ] Adds logic to the privacy center's email and phone field to allow for disabling the component when one is typed. ("XOR") 
* [ ] Adds validation such that one of email or phone number are required if both fields are optional
* [ ] Removed phone from the OOTB config.json 

### Steps to Confirm

* [ ] Spin up Privacy Center
* [ ] Update `privacy-center/config/config.json` to have `email: "optional"` instead of required, and include `phone_number: "optional"` right below each `email: "optional"`.
* [ ] Add a email and validate the phone section becomes disabled
* [ ] Add a phone and validate the email section becomes disabled
* [ ] Validate the appropriate payload is being sent in the request.

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [x] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [x] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`

### Description Of Changes

Allows for the Privacy Center's UI to match the currently functionality of the DSR engine

### CODE REVIEW

**Comment by nicolas-ethyca** [comment]
Documentation - https://github.com/ethyca/fidesdocs/pull/55

File: clients/privacy-center/components/modals/privacy-request-modal/PrivacyRequestForm.tsx [hunk 1]
```diff
@@ -239,6 +239,9 @@ const PrivacyRequestForm: React.FC<PrivacyRequestFormProps> = ({
                   onChange={handleChange}
                   onBlur={handleBlur}
                   value={values.email}
+                  disabled={Boolean(
```
Comment (NevilleS) [inline clients/privacy-center/components/modals/privacy-request-modal/PrivacyRequestForm.tsx:242]: It looks like this should use the `isDisabled` prop instead: https://chakra-ui.com/docs/components/input/props

I only noticed this because of the type change in the `PhoneInput` component, by the way - very subtle but I assume that Chakra has good reasons for that pattern

File: clients/privacy-center/components/phone-input.tsx [hunk 2]
```diff
@@ -16,7 +16,7 @@ const ReactPhoneNumberInput = dynamic(
  * Wraps the PhoneInput component from react-phone-number-input in a Chakra Input
  * with common default props.
  */
-export const PhoneInput = (props: InputProps) => (
+export const PhoneInput = (props: InputProps | { disabled?: boolean }) => (
```
Comment (NevilleS) [inline clients/privacy-center/components/phone-input.tsx:19]: This shouldn't be necessary - the `InputProps` already has a prop called `isDisabled` that'll be passed through.

File: clients/privacy-center/config/config.json [hunk 3]
```diff
@@ -8,11 +8,11 @@
     {
       "policy_key": "default_access_policy",
       "icon_path": "/download.svg",
-      "title": "Access your data",
-      "description": "We will provide you a report of all your personal data.",
+      "title": "Access your data (email or phone number)",
```
Comment (NevilleS) [inline clients/privacy-center/config/config.json:11]:
```suggestion
      "title": "Access your data",
```

This just looks odd as a default title

File: clients/privacy-center/config/config.json [hunk 4]
```diff
@@ -8,11 +8,11 @@
...
+      "description": "Please provide either an email or a phone number. We will provide you a report of all your personal data.",
       "identity_inputs": {
         "name": "optional",
-        "email": "required",
```
Comment (NevilleS) [inline clients/privacy-center/config/config.json:15]: By making the default lack any `required` inputs, we create a bit of a problem where it's possible to just submit the form blank. That throws an error message at the user that isn't very helpful.

File: src/fides/data/sample_project/privacy_center/config/config.json [hunk 5]
```diff
@@ -9,10 +9,10 @@
       "policy_key": "default_access_policy",
       "icon_path": "/assets/access.svg",
       "title": "Access your data",
-      "description": "We will provide you a report of all your personal data.",
+      "description": "Please provide either an email or a phone number. We will provide you a report of all your personal data.",
       "identity_inputs": {
         "name": "optional",
-        "email": "required",
+        "email": "optional",
```
Comment (NevilleS) [inline src/fides/data/sample_project/privacy_center/config/config.json:15]: NOTE: there's a third config.json that should be updated, which is src/fides/data/test_env/privacy_center/config/config.json.

This `test_env` config is the one used for our main regression testing, so it should be updated as well to reflect this new default behaviour

**Review by NevilleS — CHANGES_REQUESTED** [review]
Thanks for this - I did a quick review and some light testing and suggested a few changes, but we should discuss the solution for this before shipping.

In summary, couple points:

1. Update this to use the `isDisabled` prop for inputs
2. Add validation to not allow submitting without at least email or phone
3. Update the `test_env` config.json
4. Review the UX of this

Lastly here's a few screenshots of the UX for reference:

Empty
image

With email populated, phone number auto-disabled:
image
(note the subtle gray box around the numbers indicating disabled)

If you submit with just "Name" (no email or phone):
image
...then
image

File: clients/privacy-center/components/modals/privacy-request-modal/PrivacyRequestForm.tsx (hunk 1 above)
Comment (nicolas-ethyca) [reply clients/privacy-center/components/modals/privacy-request-modal/PrivacyRequestForm.tsx]: addressed

File: clients/privacy-center/config/config.json (hunk 3 above)
Comment (nicolas-ethyca) [reply clients/privacy-center/config/config.json]: addressed

**Comment by galvana** [comment]
I pulled https://github.com/ethyca/fides/issues/2451 in to the current sprint. If we're willing to wait for this to be implemented at the SaaS connector framework level we can avoid this UI change.

File: clients/privacy-center/config/config.json [hunk 6]
```diff
@@ -9,11 +9,10 @@
       "policy_key": "default_access_policy",
       "icon_path": "/download.svg",
       "title": "Access your data",
-      "description": "We will provide you a report of all your personal data.",
+      "description": "Please provide an email. We will provide you a report of all your personal data.",
       "identity_inputs": {
         "name": "optional",
```
Comment (NevilleS) [inline clients/privacy-center/config/config.json:14]: Similar to phone, I'd like us to remove `name` here as well, since it won't be used for the execution, etc.

File: clients/privacy-center/cypress/e2e/privacy-request.cy.ts [hunk 7]
```diff
@@ -13,58 +13,24 @@ describe("Privacy request", () => {
       ).as("postPrivacyRequestVerify");
     });
 
-    it("can verify phone and navigate to form", () => {
```
Comment (NevilleS) [inline clients/privacy-center/cypress/e2e/privacy-request.cy.ts:16]: Blah, this is unfortunate... can't we wire up a different config to test this? I'm pretty sure our test harnesses support swapping in different config.json?

File: src/fides/data/sample_project/privacy_center/config/config.json [hunk 8]
```diff
@@ -9,10 +9,10 @@
...
         "phone": "optional"
```
Comment (NevilleS) [inline src/fides/data/sample_project/privacy_center/config/config.json:16]: Let's update this to match the new default in the Privacy Center

File: src/fides/data/test_env/privacy_center_config/config.json [hunk 9]
```diff
@@ -12,7 +12,7 @@
       "description": "We will provide you a report of all your personal data.",
       "identity_inputs": {
         "name": "optional",
-        "email": "required",
+        "email": "optional",
         "phone": "optional"
```
Comment (NevilleS) [inline src/fides/data/test_env/privacy_center_config/config.json:16]: Same- let's have this match the privacy center defaults

**Review by NevilleS — CHANGES_REQUESTED** [review]
Couple nitpicks 😄

File: clients/privacy-center/cypress/e2e/privacy-request.cy.ts (hunk 7 above)
Comment (eastandwestwind) [reply clients/privacy-center/cypress/e2e/privacy-request.cy.ts:16]: I was able to get this working via an env variable to switch on which config.json is loaded in. I didn't find any way to do this within cypress itself.

Comment (eastandwestwind) [reply clients/privacy-center/cypress/e2e/privacy-request.cy.ts:16]: Hm, nvm, seems like using the env var is causing some unintended issues in github ci (not locally)- https://github.com/ethyca/fides/actions/runs/4632049188/jobs/8195643338?pr=2539#step:5:36.

**Comment by eastandwestwind** [comment]
Back over to you @NevilleS

File: clients/privacy-center/constants/index.ts [hunk 10]
```diff
@@ -2,8 +2,9 @@
 
 import { Config, IdentityInputs } from "~/types/config";
 import configJson from "~/config/config.json";
+import testConfigJson from "~/config/test-config.json";
 
-export const config: Config = configJson;
+export const config: Config = process.env.NEXT_PUBLIC_APP_ENV === "test" ? testConfigJson : configJson;
```
Comment (NevilleS) [inline clients/privacy-center/constants/index.ts:7]: Blah, yeah, I see the issue here... there's not a very clean & maintainable way to wire up a variety of different config.json permutations for Cypress testing, since this kind of configuration is done at build-time. 

What you've done here technically works, but at what cost? I'm not sure if the juice is worth the squeeze here because this just feels like another JSON to maintain and forget about... 🤔

**Review by NevilleS — CHANGES_REQUESTED** [review]
I'm a bit stuck on this one, sorry 🤔. The code itself is good, but I feel like the tests I asked you to add are probably not very maintainable so I kinda regret asking!

Let's discuss a bit live to decide what to keep here as I'd rather not add test debt for this edge case

File: clients/privacy-center/constants/index.ts (hunk 10 above)
Comment (NevilleS) [reply clients/privacy-center/constants/index.ts:7]: OK, so a much cleaner way to do this is to mutate the Redux store at runtime, ala this Cypress test: https://github.com/ethyca/fides/blob/main/clients/privacy-center/cypress/e2e/consent.cy.ts#L77

This requires a lot more refactoring to do, though, and clearly my scope creep has gotten out of control on this PR. I think we should just revert these new Cypress tests, manually test this change works as intended, and move forward! Sorry!

Comment (eastandwestwind) [reply clients/privacy-center/constants/index.ts]: New ticket here- https://github.com/ethyca/fides/issues/3025

**Review by NevilleS — APPROVED** [review]
Thanks for handling the back & forth here @eastandwestwind. I pushed a few more nitpicks to the branch but haven't re-tested this yet fully- going to let CI do it's thing first, but if you can just double-check we're good here then merge away.

File: clients/cypress-e2e/cypress/e2e/smoke_test.cy.ts [hunk 11]
```diff
@@ -19,7 +19,6 @@ describe("Smoke test", () => {
       cy.get("input#name").type("Jenny");
```
[the remark this answers is not in the document]
Comment (NevilleS) [reply clients/cypress-e2e/cypress/e2e/smoke_test.cy.ts]: Done!

File: clients/fides-consent/src/consent-config.json [hunk 12]
```diff
@@ -0,0 +1,31 @@
+{
+  "options": [
```
[the remark this answers is not in the document]
Comment (eastandwestwind) [reply clients/fides-consent/src/consent-config.json]: ahh, yes lots of back and forth PR switching 😓

File: clients/cypress-e2e/cypress/e2e/smoke_test.cy.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (eastandwestwind) [reply clients/cypress-e2e/cypress/e2e/smoke_test.cy.ts]: oh thank you!
